lib/index.js

```diff
@@ -80,6 +86,13 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId;
```

Comment:
If there is no userId, the track call should fail, which is what we expect. We don't want to run into a situation where the user can get unexpected results by always tracking with an anonymous userId as backup. We'd rather the userId be set explicitly so the user knows exactly what is being tracked. 

An alternative would be to require that the user call `analytics.identify` with the user id before they call `track` so that we can always rely on `.user().id()`